scalding-core/src/main/scala/com/twitter/scalding/StringUtility.scala

```diff
@@ -0,0 +1,39 @@
+package com.twitter.scalding
+
+object StringUtility {
+  // Scala's split function uses regex which is too slow thus we create this faster function
+  private def isTrailingWith(text: String, separator: String) = {
+    val result = text.foldRight((List[Boolean](), true)) {
+      case (c, (tillNow, isTrailingBefore)) =>
+        val trailing = isTrailingBefore && (c.toString == separator)
+        (tillNow ++ List(trailing), trailing)
+    }
+    result._1.reverse
+  }
+
+  private def fastSplitHelper(text: String, key: String, isTrailingAt: List[Boolean]): Seq[String] = {
+    if (text.isEmpty) {
+      Seq()
+    } else {
+      val firstIndex = text.indexOf(key)
+      if (firstIndex == -1) {
```

Comment:
Hmm.. I was trying to make the fastSplit function to be exactly the same with the regex split function. 

One thing is that, for a string trailing with the separator, regex split drops all of them. 

i.e. "a:b::" would be "a", "b". But this would gives "a", "b", ""

Would this be good enough since our usage might not include trailing separators?